### ISSUE TITLE
Broken undo in Clojure files when completion popup is shown

### ISSUE THREAD

**Issue by frangio** [Shougo/deoplete.nvim#433]
## Summary
With `ft=clojure`, when the completion popup is shown in insert mode, it breaks undo history in a pretty bad way: it makes it impossible to properly undo the changes done during insertion.

Error doesn't happen with other filetypes.

## Expected
The completion popup should have no effect on undo history.

## Environment
Neovim 0.1.7. `:CheckHealth` gives success.

## Minimal init.vim

```vim
set runtimepath+=~/path/to/deoplete.nvim/
let g:deoplete#enable_at_startup = 1
```

## Steps to reproduce after starting Neovim

 1. `:e test.clj`
 2. Insert the following text:
```
foobar
foobaz

foobar
```
 3. Begin insert mode on the last line after `foo` (`|` represents cursor): `foo|bar`.
 4. Completion popup is shown with options `foobar`, `foobaz`.
 5. Type anything, e.g. a minus sign: `foo-|bar`.
 6. Leave insert mode.
 7. Try to undo the insertion.

The inserted text remains and the next undo removes the text inserted in step 2.

**Comment by Shougo** [comment]
This is Vim/neovim bug.
I have reproduced it without deoplete.
I will report it.

**Comment by frangio** [comment]
Thank you, @Shougo.

**Comment by Shougo** [comment]
Done.

https://github.com/vim/vim/issues/1490

**Comment by Shougo** [comment]
It is fixed in Vim 8.0.341.

**Comment by Shougo** [comment]
The PR is merged.
